examples/gno.land/p/demo/uassert/uassert.gno

```diff
@@ -416,3 +419,514 @@ func NotEmpty(t TestingT, obj interface{}, msgs ...string) bool {
+		if _, ok := actual.(string); !ok {
+			return fail(t, msgs, "uassert.IsType: expected type string but got %T", actual)
+		}
+	case std.Address:
+		if _, ok := actual.(std.Address); !ok {
+			return fail(t, msgs, "uassert.IsType: expected type std.Address but got %T", actual)
+		}
+	default:
+		return fail(t, msgs, "uassert.IsType: unsupported type %T", expectedType)
+	}
+	return true
+}
+
+// Regexp asserts that a specified regexp matches matches a string.
+func Regexp(t TestingT, rx interface{}, str interface{}, msgs ...string) bool {
+	t.Helper()
+	var r *regexp.Regexp
+	switch v := rx.(type) {
+	case *regexp.Regexp:
+		r = v
+	case string:
+		r = regexp.MustCompile(v)
+	default:
+		return fail(t, msgs, "uassert.Regexp: invalid regexp type")
+	}
+
+	s, ok := str.(string)
+	if !ok {
+		return fail(t, msgs, "uassert.Regexp: invalid string type")
+	}
+
+	if !r.MatchString(s) {
+		return fail(t, msgs, "uassert.Regexp: \"%s\" does not match \"%s\"", s, r.String())
+	}
+	return true
+}
+
+// ElementsMatch asserts that two lists contain the same elements, regardless of order.
+func ElementsMatch(t TestingT, listA, listB []interface{}, msgs ...string) bool {
+	t.Helper()
+	if len(listA) != len(listB) {
+		return fail(t, msgs, "uassert.ElementsMatch: lists have different lengths")
+	}
+	elements := make(map[interface{}]int)
+	for _, item := range listA {
+		elements[item]++
+	}
+	for _, item := range listB {
+		if count, ok := elements[item]; !ok || count == 0 {
+			return fail(t, msgs, "uassert.ElementsMatch: lists do not match")
+		} else {
+			elements[item]--
+		}
+	}
+	return true
+}
+
+// Subset asserts that all elements in the subset are present in the set.
+func Subset(t TestingT, set, subset []interface{}, msgs ...string) bool {
+	t.Helper()
+	elements := make(map[interface{}]bool)
+	for _, item := range set {
+		elements[item] = true
+	}
+	for _, item := range subset {
+		if !elements[item] {
+			return fail(t, msgs, "uassert.Subset: item %v not found in set", item)
+		}
+	}
+	return true
+}
+
+// NotSubset asserts that the subset is not fully present in the set.
+func NotSubset(t TestingT, set, subset []interface{}, msgs ...string) bool {
+	t.Helper()
+	elements := make(map[interface{}]bool)
+	for _, item := range set {
+		elements[item] = true
+	}
+	for _, item := range subset {
+		if !elements[item] {
+			return true
+		}
+	}
+	return fail(t, msgs, "uassert.NotSubset: subset is fully present in set")
+}
+
+// IsInstanceOf asserts that the specified object is an instance of the given type.
+func IsInstanceOf(t TestingT, expectedType, actual interface{}, msgs ...string) bool {
+	t.Helper()
+	if !IsType(t, expectedType, actual, msgs...) {
+		return fail(t, msgs, "uassert.IsInstanceOf: %v is not an instance of %T", actual, expectedType)
+	}
+	return true
+}
+
+// Same asserts that two pointers reference the same object.
+func Same(t TestingT, expected, actual interface{}, msgs ...string) bool {
+	t.Helper()
+	if expected != actual {
```

Comment:
This won't work as expected until this bug is fixed: https://github.com/gnolang/gno/issues/2624